sys/riscv/riscv/machdep.c

```diff
@@ -814,54 +811,7 @@ sendsig(sig_t catcher, ksiginfo_t *ksi, sigset_t *mask)
 	 * than using the full capability.  Should we compare the
 	 * entire capability...?  Just pointer and bounds...?
```

Comment:
This is an interesting XXX; should probably be done based on the base

Reply:
Well, with freebsd64 we don't have a valid cap to compare against, and with purecap, I'd want to think a bit more about what the fallout would be from getting this wrong if, for example, the csp was a subset of the alternate signal stack.

Reply:
As in you'd treat the base like you currently treat the address, so if it's a subset it's still in bounds, you'd just have to be careful if the two allocations were adjacent and the base were set to the end of the first allocation (with len=0).

Reply:
Whatever we do, I don't plan to do it in this series.  We could open an issue to track this though.

Reply:
Oh, yeah, wasn't suggesting otherwise